Allow program to work without flag

enzanki-ars opened the pull request:
If env var $URL is set, there is no need for the -url.
-url will override $URL if set.

enzanki-ars committed:
Allow program to work without flag

If env var $URL is set, there is no need for the -url.
-url will override $URL if set.

jackzampolin commented on main.go:
```diff
@@ -55,6 +55,9 @@ func init() {
 
 	// Parse all flags
 	flag.Parse()
+	if slackURL == "" {
```
Can you make it quit with an error if there is no slack url passed through the flag or through a ENV variable?

maybe:

``` go
if slackURL == "" {
  slackURL = os.Getenv("CATSLACK_URL")
  if slackURL == "" {
    fmt.Println("Need to pass in slack url via the -url flag or the $SLACK_URL env variable")
    os.Exit(1)
  }
}
```

jackzampolin commented:
Once inline comment is addressed I will merge

enzanki-ars committed:
Format message better if exited due to no URL.

enzanki-ars committed:
Merge branch 'master' into get-env-var

enzanki-ars commented:
~~I added some more commits, and fixed the eventual merge conflict, but GitHub seems to be [having issues](https://status.github.com/messages) displaying them right now.~~ (Fixed)

enzanki-ars commented on main.go:
```diff
@@ -55,14 +55,19 @@ func init() {
-		return
+		fmt.Println("Must pass a working Slack integration URL using the `-url` flag")
+		fmt.Println("or by setting the $CATSLACK_URL env variable.")
+		os.Exit(1)
```
Already has this exit code though... The `return` statement already had this, though I did change it to os.Exit to stop the program with a controlled exit code.
